include/pressio/ops/tpetra/ops_level2.hpp

```diff
@@ -127,10 +127,19 @@ _product_tpetra_mv_sharedmem_vec_kokkos(const alpha_t & alpha,
 // -------------------------------
 template < class A_type, class x_type, class y_type, class alpha_t, class beta_t>
 ::pressio::mpl::enable_if_t<
-  ::pressio::all_have_traits_and_same_scalar<A_type, x_type, y_type>::value
+  // level2 common constraints
+     ::pressio::Traits<A_type>::rank == 2
+  && ::pressio::Traits<x_type>::rank == 1
+  && ::pressio::Traits<y_type>::rank == 1
+  // TPL/container specific
   && ::pressio::is_multi_vector_tpetra<A_type>::value
   && ::pressio::is_vector_tpetra<y_type>::value
   && ::pressio::is_dense_vector_teuchos<x_type>::value
+  // scalar compatibility
+  && ::pressio::all_have_traits_and_same_scalar<A_type, x_type, y_type>::value
+  && std::is_convertible<alpha_t, typename ::pressio::Traits<A_type>::scalar_type>::value
```

Comment:
where is the check for beta?

Reply:
thanks for catching this and pardon the omission... fixed on ee8991317e67d004ea8dace6fd23071e0215762c